### ISSUE TITLE
Performance issues with calcBounds

### ISSUE THREAD

**Issue by cheryldmorse** [Unidata/thredds#97]
The processing time of the FeatureDataset calcBounds function on aggregations seems to have dramatically increased from version 4.3 of netCDF Java to version 4.5. When running the following test program the calcBounds operation took 1401 ms with the netcdf-4.3.23.jar and 7008 ms with netcdf4-4.5.4.jar:
``` 
NetcdfDataset dataset = NcMLReader.readNcML(testFileName, null);
FeatureType type = FeatureDatasetFactoryManager.findFeatureType(dataset);
FeatureDataset featureDataset = findFeatureDataset(type, dataset);
featureDataset.calcBounds();
```

**Comment by rsignell-usgs** [comment]
@cheryldmorse , would be useful to include a link to the NcML file you were testing?

**Comment by cheryldmorse** [comment]
I was using this dataset in the tests: http://sos.maracoos.org/stable/agg_catalogs/weatherflow_agg_catalog.html?dataset=wflow700-agg

**Comment by cwardgar** [comment]
Unfortunately, that link does not provide the actual NcML file that you're using, nor the datasets that that file aggregates. So, I can't really reproduce your issue at this point.

**Comment by cheryldmorse** [comment]
@cwardgar The datasets and NCML file are available here: https://www.dropbox.com/s/sqfvhu5ckyrzeo3/station_700_agg.zip?dl=0
Thank you.

**Comment by cwardgar** [comment]
Okay, I've reproduced the issue. On my machine, in fact, the execution times are more like:
v4.3.24: 1692ms
v4.6.0-SNAPSHOT: 14514ms

I'm looking into it now.

**Comment by cwardgar** [comment]
Okay, I've tracked down the issue to ucar.nc2.iosp.hdf5.DataBTree

In v4.3.24, memTracker is null, and thus no messages are added to it.
In v4.6.0-SNAPSHOT, memTracker is non-null, and many expensive messages are added to it (complicated toString() implementations, mostly)

Assigning null to memTracker in v4.6.0-SNAPSHOT reduces the calcBounds() time from 14514ms to 2488ms on my computer.

I suspect that memTracker should be disabled in a non-debug context, but I don't know for sure. @JohnLCaron, please advise.

**Comment by cwardgar** [comment]
Never mind, John, I can see here [1] that MemTracker is only supposed to be used for debugging. I'll just refactor the source to check the debugTracker flag before each and every use of MemTracker.

[1] https://github.com/Unidata/thredds/blob/v4.5.4/cdm/src/main/java/ucar/nc2/iosp/hdf5/H5header.java#L222

**Comment by cwardgar** [comment]
Fixed in 4.6.0.

**Comment by JohnLCaron** [comment]
good job. i likely enabled it when debugging, and/or fixing some coverity defect. the hashmap should be null in the non-debug case.

**Comment by cwardgar** [comment]
This has also been fixed in 4.5.5:
https://github.com/Unidata/thredds/commit/be280364cccf8db7f96a82edc64585bb3c37c41b
https://github.com/Unidata/thredds/commit/c4bd0f242ec3024963759b3b2c21df8b716bd4e4

We'll try to release 4.5.5 soon.
